### PR TITLE
[libc++][doc] Update the release notes for LLVM 19.

### PR DESCRIPTION
**Pull request by mordante**
This is a preparation for the upcoming LLVM 19 release.

### CODE REVIEW

File: libcxx/docs/ReleaseNotes/20.rst [hunk 1]
```diff
@@ -59,6 +59,16 @@ Deprecations and Removals
...
 
+LLVM 20
+~~~~~~~
+
+- The ``LIBCXX_ENABLE_ASSERTIONS`` CMake variable and the ``_LIBCPP_ENABLE_ASSERTIONS`` macro that were used to enable
+  the safe mode will be removed in LLVM 20.
+
+- The C++20 synchronization library will be removed entirely in language modes prior to C++20 in LLVM 20.
+
+- The ``_LIBCPP_ENABLE_REMOVED_ALLOCATOR_CONST`` macro will no longer have an effect.
```
Comment (ldionne) [inline libcxx/docs/ReleaseNotes/20.rst:70]: Those are actually already above with TODOs, except for `_LIBCPP_ENABLE_REMOVED_ALLOCATOR_CONST`.

File: libcxx/docs/ReleaseNotes/19.rst [hunk 2]
```diff
@@ -130,9 +151,9 @@ Deprecations and Removals
   as ``std::vector<const T>`` and ``std::list<const T>``. This used to be supported as an undocumented extension.
   If you were using ``std::vector<const T>``, replace it with ``std::vector<T>`` instead. The
   ``_LIBCPP_ENABLE_REMOVED_ALLOCATOR_CONST`` macro can be defined to temporarily re-enable this extension as
-  folks transition their code. This macro will be honored for one released and ignored starting in LLVM 20.
+  to transition to update user code. This macro will be honored for one released and ignored starting in LLVM 20.
```
Comment (ldionne) [inline libcxx/docs/ReleaseNotes/19.rst:154]: `to temporarily re-enable this extension to make it easier to update user code`?

**Review by ldionne — COMMENTED** [review]
Thanks! A few comments but this looks good.

File: libcxx/docs/ReleaseNotes/19.rst [hunk 3]
```diff
@@ -55,14 +67,21 @@ Implemented Papers
...
 Improvements and New Features
 -----------------------------
 
 - The performance of growing ``std::vector`` has been improved for trivially relocatable types.
-- A lot of types are considered trivially relocatable now, including ``vector`` and ``string``.
-- The performance of ``ranges::fill`` and ``ranges::fill_n`` has been improved for ``vector<bool>::iterator``\s,
+
+- A lot of types are considered trivially relocatable now, including ``std::vector`` and ``std::string``.
+
+- The performance of ``std::ranges::fill`` and ``std::ranges::fill_n`` has been improved for ``std::vector<bool>::iterator``\s,
```
Comment (cjdb) [inline libcxx/docs/ReleaseNotes/19.rst:82]: Do we have benchmarks to show this?

File: libcxx/docs/ReleaseNotes/19.rst [hunk 4]
```diff
@@ -35,6 +35,18 @@ see the `releases page <https://llvm.org/releases/>`_.
...
+The main focus of the libc++ team has been to implement new C++20, C++23,
+and C++26 features.
+
+Experimental support for the time zone database has progressed.
+
+Work on the ranges support has progressed. See
+:ref:`ranges-status` for the current status.
+
+Work on the experimental C++17 Parallel STL has progressed. See
+:ref:`pstl-status` for the current status.
```
Comment (cjdb) [inline libcxx/docs/ReleaseNotes/19.rst:47]: I don't see these files.

File: libcxx/docs/ReleaseNotes/19.rst (hunk 3 above)
Comment (philnik777) [reply libcxx/docs/ReleaseNotes/19.rst:82]: https://github.com/llvm/llvm-project/pull/84642

File: libcxx/docs/ReleaseNotes/19.rst [hunk 5]
```diff
@@ -79,15 +98,17 @@ Improvements and New Features
 - ``std::ignore``\s ``const __ignore_t& operator=(_Tp&&) const`` was changed to
   ``const __ignore_type& operator=(const _Tp&) const noexcept`` for all language versions.
 
+- In C++23 and C++26 the number of transitive includes in several headers have been removed, improving the compilation speed.
```
Comment (Zingam) [inline libcxx/docs/ReleaseNotes/19.rst:101]: Do you mean "reduced"?

File: libcxx/docs/ReleaseNotes/19.rst (hunk 4 above)
Comment (h-vetinari) [reply libcxx/docs/ReleaseNotes/19.rst:47]: Perhaps worth mentioning here that work for P0226 has started landing (https://github.com/llvm/llvm-project/pull/89982) and now has a separate [status page](https://libcxx.llvm.org/Status/SpecialMath.html).

Comment (mordante) [reply libcxx/docs/ReleaseNotes/19.rst:47]: > I don't see these files.

I'm not sure what you mean. The generated `.html` file points to the ranges and PSTL status page.

> Perhaps worth mentioning here that work for P0226 has started landing (#89982) and now has a separate [status page](https://libcxx.llvm.org/Status/SpecialMath.html).

That was actually the reason for waiting with addressing the review comments of this PR ;-)

File: libcxx/docs/ReleaseNotes/20.rst (hunk 1 above)
Comment (mordante) [reply libcxx/docs/ReleaseNotes/20.rst:70]: Fix that. I've also copy-pasted the C++03 removal to this file.

**Review by mordante — COMMENTED** [review]
Thanks for the reviews!

**Comment by ldionne** [comment]
LGTM but this needs rebasing and to be cherry-picked to the release branch.

**Review by ldionne — APPROVED** [review]

**Comment by mordante** [comment]
/cherry-pick [e9bb4e3](https://github.com/llvm/llvm-project/commit/e9bb4e303b949f955042158567eb83000a6e7634)

**Pull request merged**